_episodes/37.md

```diff
@@ -0,0 +1,418 @@
+members of the community that aren't engineers or administrators that simply
+want to try Trino without much setup.
+
+### Release Process
+
+#### Filling out PR
+
+We've got a handy PR template that exists for all contributors to use when 
+they've submitting a pull request to Trino. Most of it is simple and
+self-explanatory - we ask you to describe what's happening, where the change is
+happening, and what type of change it is. These are for the sake of the
+reviewers, giving them some important context so they understand what's going on
+when they review the code. For simpler changes, it's not usually necessary to go
+into a ton of detail here, but it's usually nice to give a little summary if you
+think it's appropriate.
+
+The next steps are what really matter for every single PR that's going to be
+merged - the documentation and release notes for a change. These are about
+communicating to our users. Documentation refers to Trino docs, not code
+comments - if Trino users will need to be told how to use the feature you're
+changing because of how you're changing it, that means we need to have
+documentation for it. The PR template gives the options for how to go about
+this, but it's incredibly helpful to have this filled out. Similarly, we ask
+whether or not release notes are necessary for the change, and what release
+notes you would propose for your change. Generally speaking, if it needs to be
+documented, it almost always should have a release note. Even if it isn't
+documented, a release note might be a good idea - things like performance
+improvements don't require our users to change how they use Trino, but they
+won't mind knowing that something has gotten faster/better! The release process
+involves heavy editing of release notes, so it's ok for the suggested note to be
+imperfect.
+
+### What is Developer Experience?
+
+Trino is a technology that is built by developers but also heavily used by 
+developers. We want to ensure that the experience of both contributors and users
+of Trino is the best possible. To do that, we have to focus on many different
+aspects of this experience, from committing code to the CLIs and tools we offer
+for debugging queries and most importantly to building a sustainable community
+that can give answers and drive the future of the project. This is what DX is
+for Trino.
+
+### Community Metrics
+
+#### Trino-CICD Repository
+
+<https://github.com/nineinchnick/trino-cicd>
+
+Where to find it, how it works (Powered by Trino), etc...
+
+#### How understanding these metrics helps the experience of contributors
+
+By tracking this, we make sure to provide a neutral experience for all and this
+also helps us figure out where we need more resources to focus on.
+
+review process, merging process, and helps us advocate for more resources from
+companies and vendors.
+
+Are there any other metrics that are important to all of you in the contributor
```

Comment:
```suggestion

A while ago we started gathering metrics related to [the Trino GitHub repository](https://github.com/trinodb/trino).
This helped us identify issues like huge CI queue times, but most importantly, verify
that the changes we made improved things, and how much. 

In February this year, the 95th percentile of the CI queue time (not even the total run time!)
was as high as almost 7 hours. Trino uses public GitHub runners and there can only be
60 jobs running concurrently at the same time. This is a bottleneck because Trino
has extensive test coverage for the core engine and all plugins (connectors).
Because we can't increase the number of runners, we looked into doing impact analysis
to skip tests for modules not impacted by any change in a pull request.

Since April, the 95th percentile of the CI queue time is under 1 hour, even though the number of contributions
is at an all-time high.

We keep track of these selected metrics in reports we create by running queries using the Trino CLI,
saving the results in a markdown file and publishing them as static pages using GitHub pages.
The data is also gathered using third-party Trino connectors for the GitHub API and Git repositories.
There's a GitHub actions workflow running on a schedule, that spins up a Trino server, so there's no
infrastructure to maintain, except for a single S3 bucket.  
All of it is publicly available in the [nineinchnick/trino-cicd](https://github.com/nineinchnick/trino-cicd) repository.  On the right, there's a link to GitHub pages with reports.

We keep adding more reports, like:
* [flaky tests](https://nineinchnick.github.io/trino-cicd/reports/flaky/)
* [pull request activity](https://nineinchnick.github.io/trino-cicd/reports/pr/)

By being data-driven and transparent, we make sure to provide a neutral experience for all
and this also helps us figure out where we need more resources to focus on.

We're open to suggestions on what to track and which metrics to report on, so feel free
to open issues and pull requests in the repository mentioned above or start a thread on the Trino Slack.
```